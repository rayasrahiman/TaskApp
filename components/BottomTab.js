import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import React from 'react';
import AntIcon from 'react-native-vector-icons/AntDesign';
import Ionicon from 'react-native-vector-icons/Ionicons';
import FontistoIcon from 'react-native-vector-icons/Fontisto';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';

import {Colors} from '../constants/colors';
import I18n from '../languages/i18n';

export default function BottomTab({
  tab,
  dashboardPress,
  devicePress,
  schedulePress,
  notificationPress,
}) {
  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={dashboardPress} style={{alignItems: 'center'}}>
        <AntIcon
          name="appstore-o"
          style={{
            ...styles.iconSize,
            ...{
              color: tab === 'dashboard' ? Colors.darkBlue : Colors.black,
            },
          }}
        />
        <Text
          style={{
            ...styles.btnTitle,
            ...{
              color: tab === 'dashboard' ? Colors.darkBlue : Colors.black,
            },
          }}>
          {I18n.t('Dashboard')}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={devicePress} style={{alignItems: 'center'}}>
        <Ionicon
          name="water-outline"
          style={{
            ...styles.iconSize,
            ...{color: tab === 'Device' ? Colors.darkBlue : Colors.black},
          }}
        />
        <Text
          style={{
            ...styles.btnTitle,
            ...{color: tab === 'Device' ? Colors.darkBlue : Colors.black},
          }}>
          {I18n.t('Devices')}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={schedulePress} style={{alignItems: 'center'}}>
        <Ionicon
          name="ios-calendar-outline"
          style={{
            ...styles.iconSize,
            ...{color: tab === 'Schedule' ? Colors.darkBlue : Colors.black},
          }}
        />
        <Text
          style={{
            ...styles.btnTitle,
            ...{color: tab === 'Schedule' ? Colors.darkBlue : Colors.black},
          }}>
          {I18n.t('Schedule')}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={notificationPress}
        style={{alignItems: 'center'}}>
        <FontistoIcon
          name="bell"
          style={{
            ...styles.iconSize,
            ...{
              color: tab === 'Notification' ? Colors.darkBlue : Colors.black,
            },
          }}
        />
        <Text
          style={{
            ...styles.btnTitle,
            ...{
              color: tab === 'Notification' ? Colors.darkBlue : Colors.black,
            },
          }}>
          {I18n.t('Notifications')}
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: hp(4),
    shadowColor: Colors.grayishBlack,
    shadowOffset: {
      width: 0,
      height: 0.3,
    },
    shadowOpacity: 0.25,
    elevation: 5,
  },
  iconSize: {
    fontSize: hp(4),
  },
  btnTitle: {
    // color: Colors.black,
    fontSize: hp(2.5),
  },
});
