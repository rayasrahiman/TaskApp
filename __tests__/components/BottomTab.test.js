import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import BottomTab from '../../components/BottomTab';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('BottomTab', () => {
  test('should render BottomTab correctly', () => {
    const wrapper = render(<BottomTab />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
