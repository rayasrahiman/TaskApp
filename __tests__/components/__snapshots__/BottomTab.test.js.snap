// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTab should render BottomTab correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 53.5,
      "shadowColor": "#838383",
      "shadowOffset": Object {
        "height": 0.3,
        "width": 0,
      },
      "shadowOpacity": 0.25,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#000000",
            "fontSize": 53.5,
          },
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 33.5,
        }
      }
    >
      Dashboard
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#000000",
            "fontSize": 53.5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 33.5,
        }
      }
    >
      Devices
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#000000",
            "fontSize": 53.5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 33.5,
        }
      }
    >
      Schedule
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "#000000",
            "fontSize": 53.5,
          },
          Object {
            "fontFamily": "Fontisto",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 33.5,
        }
      }
    >
      Notifications
    </Text>
  </View>
</View>
`;
